worker/provisioner/lxd-broker.go

```diff
@@ -179,3 +183,29 @@ func (broker *lxdBroker) MaintainInstance(ctx context.ProviderCallContext, args
 	)
 	return err
 }
+
+func (broker *lxdBroker) writeProfiles(machineID string) ([]string, error) {
+	containerTag := names.NewMachineTag(machineID)
+	profileInfo, err := broker.api.GetContainerProfileInfo(containerTag)
+	if err != nil {
+		return nil, err
+	}
+	names := make([]string, len(profileInfo))
+	for i, profile := range profileInfo {
+		err := broker.MaybeWriteLXDProfile(profile.Name, &charm.LXDProfile{profile.Config, profile.Description, profile.Devices})
+		if err != nil {
+			return nil, err
+		}
+		names[i] = profile.Name
+	}
+	return names, nil
+}
+
+// MaybeWriteLXDProfile implements environs.LXDProfiler.
```

Comment:
Should environs.LXDProfiler be broker.LXDProfiler instead?

Reply:
Yeah, in this scenario I think so